cli/Valet/PhpFpm.php

```diff
@@ -257,4 +347,83 @@ public function validateRequestedVersion($version)
 
         return $version;
     }
+
+    /**
+     * Get FPM sock file name for a given PHP version.
+     *
+     * @param  string|null  $phpVersion
+     * @return string
+     */
+    public function fpmSockName($phpVersion = null)
+    {
+        $versionInteger = preg_replace('~[^\d]~', '', $phpVersion);
+
+        return "valet{$versionInteger}.sock";
+    }
+
+    /**
+     * Update all existing Nginx files when running a global PHP version update.
+     * If a given file is pointing to `valet.sock`, it's targeting the old global PHP version;
+     * update it to point to the new custom sock file for that version.
+     * If a given file is pointing the custom sock file for the new global version, that new
+     * version will now be hosted at `valet.sock`, so update the config file to point to that instead.
+     *
+     * @param  string  $newPhpVersion
+     * @param  string  $oldPhpVersion
+     * @return void
+     */
+    public function updateConfigurationForGlobalUpdate($newPhpVersion, $oldPhpVersion)
+    {
+        collect($this->files->scandir(VALET_HOME_PATH.'/Nginx'))
+            ->reject(function ($file) {
+                return starts_with($file, '.');
+            })
+            ->each(function ($file) use ($newPhpVersion, $oldPhpVersion) {
+                $content = $this->files->get(VALET_HOME_PATH.'/Nginx/'.$file);
+
+                if (! starts_with($content, '# Valet isolated PHP version')) {
+                    return;
+                }
+
+                if (strpos($content, $this->fpmSockName($newPhpVersion)) !== false) {
+                    info(sprintf('Updating site %s to keep using version: %s', $file, $newPhpVersion));
+                    $this->files->put(VALET_HOME_PATH.'/Nginx/'.$file, str_replace($this->fpmSockName($newPhpVersion), 'valet.sock', $content));
+                } elseif (strpos($content, 'valet.sock') !== false) {
+                    info(sprintf('Updating site %s to keep using version: %s', $file, $oldPhpVersion));
+                    $this->files->put(VALET_HOME_PATH.'/Nginx/'.$file, str_replace('valet.sock', $this->fpmSockName($oldPhpVersion), $content));
+                }
+            });
+    }
+
+    /**
+     * Get a list of all PHP versions currently serving "isolated sites" (sites with custom Nginx
+     * configs pointing them to a specific PHP version).
```

Comment:
Oops--I assume this is also the PHP version being used globally? My fault, if so. If so, here's an updated text for it:

```suggestion
     * Get a list including the global PHP version and allPHP versions currently serving "isolated sites" (sites with
     * custom Nginx configs pointing them to a specific PHP version).
```